GitCommands/Git/GitGpgController.cs

```diff
@@ -0,0 +1,297 @@
+        /// <returns>Full string coming from GPG analysis on current revision.</returns>
+        string GetCommitVerificationMessage();
+
+        /// <summary>
+        /// Obtain the tag status on current revision.
+        /// </summary>
+        /// <returns>Enum value that indicate if current git revision has one tag with good signature, one tag with bad signature or more than one tag.</returns>
+        GitGpgController.TagStatus CheckTagSign();
+
+        /// <summary>
+        /// Accessor for Revision property.
+        /// </summary>
+        GitRevision Revision { get; set; }
+
+        /// <summary>
+        /// Accessor for Module property.
+        /// </summary>
+        IGitModule Module { get; set; }
+
+        /// <summary>
+        /// Obtain the number of tag on current git revision.
+        /// </summary>
+        /// <returns>Returns the number of annotated or signed tag on current revision.</returns>
+        int NumberOfTag { get; }
+
+        /// <summary>
+        /// Obtain the tag verification message for all the tag in current git revision 
+        /// </summary>
+        /// <returns>Full concatenated string coming from GPG analysis on all tags on current git revision.</returns>
+        string TagVerifyMessage { get; }
+    }
+
+
+    public class GitGpgController : IGitGpgController
+    {
+        private IGitModule _module;
+        private GitRevision _revision;
+
+        /* Commit GPG status */
+        private const string GOOD_SIGN = "G";
+        private const string BAD_SIGN = "B";
+        private const string UNK_SIGN_VALIDITY = "U";
+        private const string EXPIRED_SIGN = "X";
+        private const string EXPIRED_SIGN_KEY = "Y";
+        private const string REVOKED_KEY = "R";
+        private const string MISSING_PUB_KEY = "E";
+        private const string NO_SIGN = "N";
+
+
+        /* Tag GPG status */
+        private const string _goodSignature = "GOODSIG";
+        private const string _validTagSign = "VALIDSIG";
+        private const string _noTagPubKey = "NO_PUBKEY";
+
+
+        /// <summary>
+        /// Obtain the number of tag on current git revision.
+        /// </summary>
+        /// <returns>Returns the number of annotated or signed tag on current revision.</returns>
+        public int NumberOfTag { get; private set; }
+
+        /// <summary>
+        /// Obtain the tag verification message for all the tag in current git revision 
+        /// </summary>
+        /// <returns>Full concatenated string coming from GPG analysis on all tags on current git revision.</returns>
+        public string TagVerifyMessage { get; private set; }
+
+
+        public enum CommitStatus
+        {
+            GoodSignature = 0,
+            SignatureError = 1,
+            MissingPublicKey = 2,
+            NoSignature = 3
+        };
+
+        public enum TagStatus
+        {
+            OneGood = 0,
+            OneBad = 1,
+            Many = 2,
+            NoPubKey = 3
+        };
```

Comment:
Please move both enums to the outer namespace

Reply:
ok I'll do it.